src/main/daml/Daml/Finance/Instrument/Generic/Election.daml

```diff
@@ -54,7 +54,13 @@ template Election
       view = Election.View with
         id; description; amount; elector; counterparty; electorIsOwner; observers; provider
         instrument; claim
-      archive' = archive . fromInterfaceContractId @Election
+      apply self Election.Apply{observableCids; exercisableCid} = do
+        let v = view $ toInterface @Election.I this
+        (newInstrumentCid, effects) <- exercise exercisableCid Election.ApplyElection with
+          observableCids
+          electionCid = toInterfaceContractId self
+        archive self -- this is needed as we can't write postconsuming choices on interfaces
```

Comment:
After this refactoring we can basically get rid of the `archive` interface method on the election I guess

Reply:
oh, you did it already :) thanks, makes sense!